src/utils.ts

```diff
@@ -370,3 +371,60 @@ export async function findBlockAtOrOlder(provider: providers.Provider, desiredLo
   }
   return fromBlockNumber;
 }
+
+export type UpdateDepositDetailsMessageType = {
+  UpdateDepositDetails: [
+    {
+      name: "depositId";
+      type: "uint32";
+    },
+    { name: "originChainId"; type: "uint256" },
+    { name: "updatedRelayerFeePct"; type: "int64" },
+    { name: "updatedRecipient"; type: "address" },
+    { name: "updatedMessage"; type: "bytes" }
```

Comment:
Is this part of the ABI or something that would be exposed by TypeChain?

Reply:
Nope, AFAICT this is not exposed by TypeChain because this struct is not part of the ABI... There is a constant in the `SpokePool.sol` contract
```solidity
    bytes32 public constant UPDATE_DEPOSIT_DETAILS_HASH =
        keccak256(
            "UpdateDepositDetails(uint32 depositId,uint256 originChainId,int64 updatedRelayerFeePct,address updatedRecipient,bytes updatedMessage)"
        );
```
but TypeChain can't extract types for EIP-712 compliant data it seems from that alone.